tests/test_registrations/test_views.py

```diff
@@ -0,0 +1,476 @@
+        reg = self.node.register_node(DEFAULT_METASCHEMA, Auth(self.user), '', None)
+        url = reg.web_url_for('node_register_page')
+        res = self.app.get(url, auth=self.user.auth)
+        assert_equal(res.status_code, http.OK)
+
+    def test_non_admin_can_view_node_register_page(self):
+        non_admin = AuthUserFactory()
+        self.node.add_contributor(
+            non_admin,
+            permissions.DEFAULT_CONTRIBUTOR_PERMISSIONS,
+            auth=Auth(self.user),
+            save=True
+        )
+        reg = RegistrationFactory(project=self.node)
+        url = reg.web_url_for('node_register_page')
+        res = self.app.get(url, auth=non_admin.auth)
+        assert_equal(res.status_code, http.OK)
+
+    def test_is_public_node_register_page(self):
+        self.node.is_public = True
+        self.node.save()
+        reg = RegistrationFactory(project=self.node)
+        reg.is_public = True
+        reg.save()
+        url = reg.web_url_for('node_register_page')
+        res = self.app.get(url, auth=None)
+        assert_equal(res.status_code, http.OK)
+
+    @mock.patch('framework.tasks.handlers.enqueue_task', mock.Mock())
+    def test_register_template_page_backwards_comptability(self):
+        # Historically metaschema's were referenced by a slugified version
+        # of their name.
+        reg = self.draft.register(
+            auth=Auth(self.user),
+            save=True
+        )
+        url = reg.web_url_for(
+            'node_register_template_page',
+            metaschema_id=_name_to_id(self.meta_schema.name),
+        )
+        res = self.app.get(url, auth=self.user.auth)
+        assert_equal(res.status_code, http.OK)
+
+    def test_register_template_page_redirects_if_not_registration(self):
+        url = self.node.web_url_for(
+            'node_register_template_page',
+            metaschema_id=self.meta_schema._id,
+        )
+        res = self.app.get(url, auth=self.user.auth)
+        assert_equal(res.status_code, http.FOUND)
+
+
+class TestDraftRegistrationViews(RegistrationsTestBase):
+
+    def tearDown(self):
+        super(TestDraftRegistrationViews, self).tearDown()
+        DraftRegistration.remove()
+
+    def test_submit_draft_for_review(self):
+        url = self.draft_api_url('submit_draft_for_review')
+        res = self.app.post_json(
+            url,
+            self.embargo_payload,
+            auth=self.user.auth
+        )
+        assert_equal(res.status_code, http.ACCEPTED)
+        data = res.json
+        assert_in('status', data)
+        assert_equal(data['status'], 'initiated')
+
+        self.draft.reload()
+        assert_is_not_none(self.draft.approval)
+        assert_equal(self.draft.approval.meta, {
+            u'registration_choice': unicode(self.embargo_payload['registrationChoice']),
+            u'embargo_end_date': unicode(self.embargo_payload['embargoEndDate'])
+        })
+
+    def test_draft_before_register_page(self):
```

Comment:
The test name makes it a bit unclear about what this test is supposed to do. A more descriptive name might be helpful.

Sample clarification questions: Are we just verifying the page exists? Or do we also (as part of the test) need to check that the page contains particular expected text (like node title)?